src/Mapping/Driver/AttributeDriver.php

```diff
@@ -117,58 +107,32 @@ public function loadClassname(string $classname): array
     private function getClassMetadataForClassname(
         string $classname
     ): ?ClassMetadata {
-        $reader = new FileCacheReader(
-            new AnnotationReader(),
-            $this->cachePath,
-            $this->debug
-        );
-
         $reflClass = new \ReflectionClass($classname);
-        /** @var Annotations\Entity|null */
-        $classAnnotation = $reader->getClassAnnotation(
-            $reflClass,
-            Annotations\Entity::class
-        );
+        $classAttribute = $this->getClassAttribute($reflClass, Attributes\Entity::class);
 
-        if (!$classAnnotation) {
+        if (!$classAttribute) {
             return null;
         }
 
         $attributeList = [];
         $relationList = [];
         foreach ($reflClass->getProperties() as $property) {
             // manage attributes
-            /** @var Annotations\Attribute|null */
-            $propertyAnnotation = $this->getPropertyAnnotation(
-                $reader,
-                $property,
-                'Attribute'
-            );
-
-            if ($propertyAnnotation) {
-                $isId = $this->getPropertyAnnotation($reader, $property, 'Id');
+            $propertyAttribute = $this->getPropertyAttribute($property, Attributes\Attribute::class);
+            if ($propertyAttribute) {
+                $propertyIsAnIdentifier = $this->getPropertyAttribute($property, Attributes\Id::class);
+                $propertyIsAnIdentifier = $propertyIsAnIdentifier instanceof Attributes\Id;
```

Comment:
Don't use the same name for something that is not the same.

Either do 
```suggestion
                $propertyIsAnIdentifier = $this->getPropertyAttribute($property, Attributes\Id::class) instanceof Attributes\Id;
```

or
```suggestion
                $propertyAttribute = $this->getPropertyAttribute($property, Attributes\Id::class);
                $propertyIsAnIdentifier = $propertyAttribute instanceof Attributes\Id;
```